### ISSUE TITLE
Cannot read property 'add' of undefined in AudioPlayer.playBroadcast (w/ ytdl-core)

### ISSUE THREAD

**Issue by chrisxclash** [discordjs/discord.js#2699]
**Please describe the problem you are having in as much detail as possible:**

I'm currently trying to use a stream (`ytdl-core` in this case, as I've seen in multiple docs). I'm able to get a local mp3 to play just fine, but I can't for the life of me get past this error with streams no matter what I've tried. I fell back to using the example shown in the `PlayInterface` itself just to ensure it wasn't bad data from my end.

This is my first time messing with streams, so if this is an issue with `ytdl-core` let me know and I can try posting it as an issue over there.

**Include a reproducible code sample here, if possible:**

```js
async play(message) {
  try {
    const connection = await message.member.voiceChannel.join();
    const stream = ytdl('https://www.youtube.com/watch?v=ZlAU_w7-Xp8', { quality: 'highestaudio' })
    connection.play(stream);
  } catch (e) {
    console.error(e);
  }
}
```

**Further details (Stack Trace):**

```js
TypeError: Cannot read property 'add' of undefined
    at AudioPlayer.playBroadcast (/node_modules/discord.js/src/client/voice/player/AudioPlayer.js:20:27)
    at VoiceConnection.play (/node_modules/discord.js/src/client/voice/util/PlayInterface.js:64:26)
```

**Versions**
- `ffmpeg`: v4.0.2
- `node-opus`: v0.3.0
- `discord.js`: v12.0.0-dev
- `node.js`: v8.9.1
- Operating system: MacOS High Sierra (10.13.6)

**Feedback**
- Priority this issue should have: Unsure; Haven't seen any other occurances on Google/issues
- [ ] I found this issue while running code on a __user account__
- [x] I have also tested the issue on latest master, commit hash: 4ae58f6

**Comment by NbOpposite** [comment]
Line 62-65 of PlayInterface.js
```js
    if (resource instanceof Broadcast) {
      if (!this.player.playBroadcast) throw new Error('VOICE_PLAY_INTERFACE_NO_BROADCAST');
      return this.player.playBroadcast(resource, options);
    }
```

I've walked trough the codepaths leading to that error, and I can't for the life of me understand how that first if-statement gets entered. I can play ytdl streams just fine, did you test the reproducible code to verify that it actually produces the error? If it does, I'll let someone more competent deal with this. If it doesn't produce the error, come on over to https://discord.gg/bRCvFy9 and I'll see if I can help spot the error in your original code

**Comment by chrisxclash** [comment]
Yeah I've poked around in it for a bit as well, but I wasn't able to get super far with it based on my limited exposure with the codebase.

Yeah, the reproducible code that I entered above is one of the 30 different formats that I've tried over the past 3 hours to get it working. I was originally on v11.3.2 and wasn't able to get _anything_ to play. The v12-beta is at least giving me some results, but the error above is the one I've been stuck on for a while now.

I'm actually already in the discord and posted about it in the #lib-voice-not-ytdl channel (Ducky Quackerson), which I don't even know if that's the right channel given the name, lol.
